fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_IdentPage.js

```diff
@@ -78,40 +118,81 @@ class CDUIdentPage {
         let storedRoutesRunwaysCell = "";
         let storedWaypointsNavaidsCell = "";
         let storedDeleteCell = "";
+        let secondaryDBSubLine = "";
+        let secondaryDBTopLine = "";
         if ((stored.routes + stored.runways + stored.waypoints + stored.navaids) > 0) {
             storedTitleCell = "STORED\xa0\xa0\xa0\xa0";
             storedRoutesRunwaysCell = `{green}${stored.routes.toFixed(0).padStart(2, '0')}{end}{small}RTES{end}\xa0{green}${stored.runways.toFixed(0).padStart(2, '0')}{end}{small}RWYS{end}`;
             storedWaypointsNavaidsCell = `{green}{big}${stored.waypoints.toFixed(0).padStart(2, '0')}{end}{end}{small}WPTS{end}\xa0{green}{big}${stored.navaids.toFixed(0).padStart(2, '0')}{end}{end}{small}NAVS{end}`;
-            storedDeleteCell = confirmDeleteAll ? '{amber}CONFIRM DEL*{end}' : '{cyan}DELETE ALL}{end}';
+            storedDeleteCell = (confirmType == ConfirmType.DeleteStored) ? '{amber}CONFIRM DEL*{end}' : '{cyan}DELETE ALL}{end}';
 
             // DELETE ALL
             mcdu.onRightInput[4] = () => {
-                if (confirmDeleteAll) {
+                if (confirmType == ConfirmType.DeleteStored) {
                     const allDeleted = mcdu.dataManager.deleteAllStoredWaypoints();
                     if (!allDeleted) {
                         mcdu.setScratchpadMessage(NXSystemMessages.fplnElementRetained);
                     }
                     CDUIdentPage.ShowPage(mcdu);
                 } else {
-                    CDUIdentPage.ShowPage(mcdu, true);
+                    CDUIdentPage.ShowPage(mcdu, ConfirmType.DeleteStored);
                 }
             };
         }
 
+        //Potential future message in case we switch to a newer version of the FMS
+        if (confirmDataBaseSwitch) {
+            secondaryDBTopLine = (confirmType == ConfirmType.SwitchDataBase) ? "{red}{small} " + calculateSecDate(date) + "{end}" : "\xa0SECOND NAV DATA BASE";
+            secondaryDBSubLine = (confirmType == ConfirmType.SwitchDataBase) ? '{amber}{CANCEL   SWAP  CONFIRM*{end}' : "{small}{" + calculateSecDate(date) + "{end}[color]cyan";
+        }
+        else {
```

Comment:
This should really show the start and end date of the current cycle IMO, since that's what you're switching to.
```suggestion
        // H4+ only confirm prompt
        if (confirmDataBaseSwitch) {
            secondaryDBTopLine = (confirmType === ConfirmType.SwitchDataBase) ? "{red}{small} " + calculateSecDate(date) + "{end}" : "\xa0SECOND NAV DATA BASE";
            secondaryDBSubLine = (confirmType === ConfirmType.SwitchDataBase) ? '{amber}{CANCEL    SWAP CONFIRM*{end}' : "{small}{" + calculateSecDate(date) + "{end}[color]cyan";
        } else {
```

Reply:
To confirm, you want me to display the Primary database date instead of the secondary in the swap message because we don't have the secondary database available yet ?
I personally wanted to act as if we were switching to the secondary database and use its date even though we were not switching in the end.

Reply:
Also, note that the date display when you swap should include also the year, not just day and month. But that would require me to change the date functions more in depth. 
Are you ok we do not put the year yet ? I skipped it because it's also for a newer version of the FMC.

Reply:
implemented suggested change, didn't change anything related to which date should be displayed.